spec/asyncapi.md

```diff
@@ -1159,9 +1181,11 @@ Describes a message received on a given channel and operation.
 Field Name | Type | Description
 ---|:---:|---
 <a name="messageObjectMessageId"></a>messageId | `string` | Unique string used to identify the message. The id MUST be unique among all messages described in the API. The messageId value is **case-sensitive**. Tools and libraries MAY use the messageId to uniquely identify a message, therefore, it is RECOMMENDED to follow common programming naming conventions.
+<a name="messageObjectType"></a>type | `string` | The type of the message. Should be set if message is used in a [Operation](#operationsObject) using request/reply pattern. Allowed values are: `generic`, `request`, `response`. Default value is `generic`.
 <a name="messageObjectHeaders"></a>headers | [Schema Object](#schemaObject) &#124; [Reference Object](#referenceObject) | Schema definition of the application headers. Schema MUST be of type "object". It **MUST NOT** define the protocol headers.
 <a name="messageObjectPayload"></a>payload | `any` | Definition of the message payload. It can be of any type but defaults to [Schema object](#schemaObject). It must match the schema format, including encoding type - e.g Avro should be inlined as either a YAML or JSON object NOT a string to be parsed as YAML or JSON.
 <a name="messageObjectCorrelationId"></a>correlationId | [Correlation ID Object](#correlationIdObject) &#124; [Reference Object](#referenceObject) | Definition of the correlation ID used for message tracing or matching.
+<a name="messageObjectReplyTopic"></a>replyTopic | [ReplyTopic Object](#replyTopicObject) &#124; [Reference Object](#referenceObject) | Should be set if message is used in a [Operation](#operationsObject) using request/reply pattern. If a application creates a inbox for all reponses. The inbox address have to be given in replyTopic field.
```

Comment:
A channel can have multiple messages and each message can have the reply topic in a different positions (header, payload).
Ok this is not very likely and a bad designt but this is nothing we should judge.

I renamed it to replyChannel

Reply:
In v3, we can have now as many operations as we want, even on the same channel. I think it would be more readable if we remove this from here and just leave this info in the `operation.reply.channel` field. We already have it:

```yaml
operations:
  pingPong:
    action: send
    channel: 
      $ref: '#/channels/ping'
    reply:
      channel: 
        $ref: '#/channels/pong'
```

Reply:
That's why I asked you on Slack to add the `messages` field to the operation object, so we can select which messages this operation applies to. Something like this:

```yaml
operations:
  pingPongV1:
    action: send
    channel: 
      $ref: '#/channels/ping'
    messages:
      - 'pingV1'
    reply:
      channel: 
        $ref: '#/channels/pong'
  pingPongV2:
    action: send
    channel: 
      $ref: '#/channels/ping'
    messages:
      - 'pingV2'
    reply:
      channel: 
        $ref: '#/channels/pong'
```

This makes me wonder if we should also have a `messages` field in the `reply` object, so we can specify that, even though the reply will be on the given channel, it will only use a subset of the allowed messages of this channel. Something like this:

```yaml
operations:
  pingPongV1:
    action: send
    channel: 
      $ref: '#/channels/ping' # I'm assuming the ping channel can have pingV1 and pingV2 messages.
    messages:
      - 'pingV1'
    reply:
      channel: 
        $ref: '#/channels/pong' # I'm assuming the pong channel can have pongV1 and pongV2 messages.
      messages:
        - pongV1
  pingPongV2:
    action: send
    channel: 
      $ref: '#/channels/ping'
    messages:
      - 'pingV2'
    reply:
      channel: 
        $ref: '#/channels/pong'
      messages:
        - pongV2
```

This should also answer @jonaslagoni question: https://github.com/asyncapi/spec/pull/847#discussion_r992453627.

Reply:
The message field was there once. But after discussion with @jonaslagoni in chat i removed it.
Because i aggreged that the also can be multiple channels pointing to the same topic.

Like this:
```yaml
channels:
  pingV1:
    address: /
    messages:
      pingV1:
        $ref: '#/components/messages/pingV1'
  pongV1:
    address: /
    messages:
      pongV1:
        $ref: '#/components/messages/pongV1'
  pingV2:
    address: /
    messages:
      pingV2:
        $ref: '#/components/messages/pingV2'
  pongV2:
    address: /
    messages:
      pongV2:
        $ref: '#/components/messages/pongV2'

operations:
  pingPongV1:
    action: send
    channel: 
      $ref: '#/channels/pingV1' # I'm assuming the ping channel can have pingV1 and pingV2 messages.
    replyChannel: 
      $ref: '#/channels/pongV1' # I'm assuming the pong channel can have pongV1 and pongV2 messages.
  pingPongV2:
    action: send
    channel: 
      $ref: '#/channels/pingV2'
    replyChannel:
      $ref: '#/channels/pongV2'

```

Reply:
Yeah, that's a good point. Let's keep it out of this PR then. And yeah, if we're not going to have a `messages` property here, I'd make the `messages` key in channels an array instead.

Reply:
reply.channel is ok. is a little more complex but more future save.